include/exadg/operators/operator_base.cpp

```diff
@@ -541,6 +541,12 @@ OperatorBase<dim, Number, n_components>::apply_inverse_block_diagonal(VectorType
   // matrix-free
   if(this->data.implement_block_diagonal_preconditioner_matrix_free)
   {
+    AssertThrow((evaluate_face_integrals() and data.use_cell_based_loops) or
```

Comment:
It seems as if you could simply write

```c++
if(evaluate_face_integrals())
{
  AssertThrow(data.use_cell_based_loops, ...);
}
```
as already suggested here https://github.com/exadg/exadg/pull/380#issuecomment-1463802569. This would be much simpler to understand in my opinion.